### ISSUE TITLE
Updates aren't easy for existing users

### ISSUE THREAD

**Issue by jamesefhawkins** [PostHog/posthog#170]
**Is your feature request related to a problem? Please describe.**
We are launching new features every day.

We have no easy way to let users just roll onto a new version of the platform to take advantage of these.

There are multiple ways users may have deployed - Heroku, Docker or from source.

**Describe the solution you'd like**
I'd really like someone to help out with how this should work.

**Describe alternatives you've considered**
Recompile the whole thing from scratch and migrate the database... a ton of work.

**Comment by allan-simon** [comment]
on this, the best upgrade experience I ever got was with gitlab and their `omnibus` package,  I `sudo apt-get update`  `sudo apt-get upgrade` , and that's all , the package take care of upgrading , rollbacking if things go south, to the point I do it blindly. Of course gitlab have a lot more workforce than the two of you ^^'

The other alternative for docker based installation is to have your docker file that has startup check the database for migration and upgrade and then launch the service ? (other possibility: in our company we have a dedicated docker image that do specifically this, in order to launch it only once)

**Comment by timgl** [comment]
@allan-simon Thanks for that, will have a look at omnibus!

RE the docker thing, that means that each time the docker image is started it just pings the git repository for new code and runs the migrations?

@jamesefhawkins For Heroku, I reckon we might have to do a separate Django app that we host that people can connect with their Heroku accounts. Then whenever there's an upgrade we can just deploy a new version for them.

**Comment by allan-simon** [comment]
@timgl  for this basically here's how we deploy metabase ( https://github.com/metabase/metabase )

1. we have a docker cluster,  (using ECS if you're familiar with AWS stack) that pull metabase:latest  image
2. when there's a new version of metabase, we kill the old instances, it pop new one, and they automatically donwload the newest :latest image, and I think metabase people have configured the image to run db migration etc. as part of the startup process 

so if you docker image and tagging process follow the same steps, you should be fine , I don't think it's needed to check the git repository etc.

**Comment by Twixes** [comment]
Broadly superseded by #2789.
